### ISSUE TITLE
Reinforce corner walls

### ISSUE THREAD

**Issue by Loobinex** [OpenDungeons/OpenDungeons#1138]
Please see the screenshot below. Notice the workers have reinforced everything they can, but the corner tiles remain neutral. It would look a bit prettier if the corner tiles would also be filled when two adjacent slaps are reinforced.

Even when all tiles are reinforced, the little corner stone is missing.

![reinforcedcorner](https://cloud.githubusercontent.com/assets/13840686/14035230/22383b14-f22c-11e5-8c03-51c005ddf691.jpg)

**Comment by hwoarangmy** [comment]
This is known and has already been discussed. But doing so is not straight forward and would change the gameplay (as heroes wanting to dig to your dungeon would not be able to dig trough claimed tiles). IMHO, it is a won't fix. The best would be to find a model for claimed tiles that looks better when claimed tiles are next to not claimed ones.

@OpenDungeons/developers WDYT ?

**Comment by Danimal696** [comment]
the only i can think of, is making 3 new tiles (texture only, no new mesh) with 1,2 and 4 small "walled" corners, while the rest is dirt.

**Comment by Loobinex** [comment]
@hwoarangmy Are you sure you want heroes to dig through those tiles? In the screenshot above you see three green arrows:
1) The tile completely surrounded by reinforced walls is totally inaccessible in all cases.
2) The tile on the right is actually reinforced so cannot be dug through, still it has no cornerstone.
3) The southern tile can currently be dug through by heroes, but what is the upside? It won't cause a breach, it slows down the heroes a bit and it gives a not so nice look. In other words, why is it fun for the player for those tiles to be dug through by heroes?

**Comment by hwoarangmy** [comment]
> why is it fun for the player for those tiles to be dug through by heroes?

That's not my point. The way I see it, reinforcing thoses walls will not be straight forward (code wise) and will introduce some special behaviours (auto-claim tiles when some surrounding tiles have been claimed - unclaiming when surrounding tiles have been unclaimed).
Moreover, it will introduce some build orders where you can use auto claim to claim tiles in your dungeon (in your screenshot, for example, if the tiles around the portal are auto claimed and you build a room bigger than 3x3 on its left, you will get some free claimed tiles

**Comment by Loobinex** [comment]
It will not provide an advantage to try and have additional corner walls to more quickly claim ground. This is because digging walls should take more time than digging dirt, which offsets the time gained.

That said I certainly agree that it's not straightforward to implement for a very trivial thing.

**Comment by Bertram25** [comment]
I must say that would one "bug" remains in OD, it would be that one. That's a lot of work code-wise + testing for a trivial detail.
On another hand, if we had tiles looking better in that situation, it would be evenn better, as no additional code would be involved. :)

**Comment by Danimal696** [comment]
so the new tiles textures would be the best fix, fine by me, but there is a lot of things to do before that, ill do it as a low priority thing.

**Comment by Loobinex** [comment]
If somebody can point me towards the current textures, I might be able to provide this one.

**Comment by Danimal696** [comment]
which graphic suite do you use?

**Comment by Loobinex** [comment]
At the moment? Paint. ;).

For something like this? That would depend on the format of the texture files.

**Comment by Danimal696** [comment]
Can you use gimp or photoshop?

**Comment by Loobinex** [comment]
I've worked with photoshop in the past. I've never used GIMP.

**Comment by Danimal696** [comment]
in the game folder, Materials->Textures ; dirt.png should be your base,
DungeonClaimedWall1111 is the closest to what we need, but the lower part being dirt instead of wall and no "flower"; each 1 in the name stands for one corner being true (near a claimed wall) so you need to create four variations from 1 to 4 corners each. Its easier once you see it.

**Comment by hwoarangmy** [comment]
> I've worked with photoshop in the past. I've never used GIMP.

GIMP is a photoshop equivalent but free and open source. If you never tried, it is worth testing ^^

**Comment by Loobinex** [comment]
I believe I made what is needed. This is the result:
Before:
![before](https://cloud.githubusercontent.com/assets/13840686/14068628/ae1b9536-f489-11e5-8520-f804a0a33779.jpg)

After:
![after](https://cloud.githubusercontent.com/assets/13840686/14068630/b54c7744-f489-11e5-9db4-f6890381456a.jpg)

Here are the files:
[reinforcedcorners.zip](https://github.com/OpenDungeons/OpenDungeons/files/191521/reinforcedcorners.zip)

**Comment by Danimal696** [comment]
nice

**Comment by hwoarangmy** [comment]
@Loobinex The problem with your texture is that it would fit a special dirt tile (around claimed walls). ATM, we use configurable tilesets. Using special tiles like the ones you made would require to add special cases in the tileset (dirt tile linked to claimed tile). It is possible but quite some work (and possible bugs) for a not mandatory feature IMHO.
When I was talking about changing models, I was thinking about finding a tile model that would not look weird if next to a dirt tile.

**Comment by Danimal696** [comment]
he did pretty well for the dirt corners, cant they be declared in the tileset?

**Comment by Loobinex** [comment]
@hwoarangmy No problem. Feel free to keep the files for a possible future use. I appreciate this is merely a cosmetic issue and gameplay has to come first.

**Comment by Danimal696** [comment]
your addition is pretty welcome, im sure we wil manage the technical issues of it

**Comment by Danimal696** [comment]
@hwoarangmy when i try to create dirtCorner in the tileset (just a copy of dirtFull, i dont actually use it in any declaration) the game doesnt load and the log shows this:

```
(16:17:32) (ConfigManager) [NORMAL] Load Tilesets file: ./config/tilesets.cfg
(16:17:32) (ConfigManager) [CRITICAL] (ConfigManager.cpp:1072) Expecting TileSet tag but got=[dirtCorner]
(16:17:32) (ConfigManager) [CRITICAL] (ConfigManager.cpp:128) Couldn't read loadTilesets
```

Its impossible to add a new kind of tile? is it hardcoded for just the existant ones?

**Comment by hwoarangmy** [comment]
> Its impossible to add a new kind of tile? is it hardcoded for just the existant ones?

Yes, use of the different kind of tiles is hardcoded. If we want to integrate such kind of tiles, we would need to declare all the possible tiles with all possible neighboor tiles (ground and full). ATM, to reduce combinations, we declare linked and not linked tiles (per type) and the tile types that link together.
That means that at the moment, we have:
6 \* 2 \* 2^4 = 192 + links per tileset.

If we want to declare all combinations, 12^4 = 20736 per tileset.

That would make the tileset file pretty unreadable and might also have an impact at runtime.

Of course, we can also define a specific tile type (for example full dirt link and declare how dirt tiles are linked to other tile types. But IMHO, it is not a solution because this kind of problem can be seen on another tile types.

I think the best solution would be to change a bit the claimed tile model to make it not look weird when next to a dirt tile. For example, by displaying a checkboard (dirt/claimed) on the upside of the tile.

**Comment by Danimal696** [comment]
i cant imagine that, can you put a dirty quick sample?

**Comment by hwoarangmy** [comment]
Something like that:

![ae1b9536-f489-11e5-8520-f804a0a33779](https://cloud.githubusercontent.com/assets/7493706/14091960/a392831c-f543-11e5-9eb1-f9f852cce668.jpg)

**Comment by Danimal696** [comment]
You have to admit it looks bad no matter how you look it... i will think for something in the long run..

**Comment by hwoarangmy** [comment]
> You have to admit it looks bad no matter how you look it...

Yep. That's why artists are needed ;-)
Note that it is only one solution, there might be other options. However, I think we should stick on this kind of solution (ie not changing dirt tiles depending on other tile types)
